input/system/azure/system.go

```diff
@@ -0,0 +1,302 @@
+package azure
+
+import (
+	"context"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/monitor/azquery"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/cosmosforpostgresql/armcosmosforpostgresql"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/postgresql/armpostgresqlflexibleservers/v4"
+	"github.com/pganalyze/collector/state"
+	"github.com/pganalyze/collector/util"
+)
+
+// GetSystemState - Gets system information about an Azure instance
+func GetSystemState(ctx context.Context, server *state.Server, logger *util.Logger) (system state.SystemState) {
+	config := server.Config
+	system.Info.Type = state.AzureDatabaseSystem
+	if config.AzureSubscriptionID == "" {
+		server.SelfTest.MarkCollectionAspectWarning(state.CollectionAspectSystemStats, "unable to collect system stats")
+		server.SelfTest.HintCollectionAspect(state.CollectionAspectSystemStats, "Config value azure_subscription_id is required to collect system stats.")
+		return
+	}
+
+	credential, err := getAzureCredential(config)
+	if err != nil {
+		server.SelfTest.MarkCollectionAspectError(state.CollectionAspectSystemStats, "error getting credential: %v", err)
+		server.SelfTest.HintCollectionAspect(state.CollectionAspectSystemStats, "Make sure the managed identity is assigned to the collector VM.")
+		logger.PrintError("Azure/System: Encountered error getting credential: %v\n", err)
+		return
+	}
+
+	var resourceID string
+
+	// Server info: Flexible Server
+	clientFactory, err := armpostgresqlflexibleservers.NewClientFactory(config.AzureSubscriptionID, credential, nil)
+	if err != nil {
+		server.SelfTest.MarkCollectionAspectError(state.CollectionAspectSystemStats, "error making a Flexible Server client: %v", err)
+		logger.PrintError("Azure/System: Failed to make a Flexible Server client: %v\n", err)
+		return
+	}
+	// Search a server from the list
+	pager := clientFactory.NewServersClient().NewListPager(nil)
+	for pager.More() {
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			logger.PrintError("Azure/System: Failed to advance page of Flexible Server list: %v\n", err)
+			break
+		}
+		for _, v := range page.Value {
+			if v.ID != nil {
+				resourceIDParts := strings.Split(*v.ID, "/")
+				resourceGroup := resourceIDParts[4]
+				resourceType := resourceIDParts[7]
+				resourceName := resourceIDParts[8]
+
+				if config.AzureDbServerName == resourceName {
+					system.Info.Azure = &state.SystemInfoAzure{
```

Comment:
FYI: state/system.go file is a good place to check and discuss if we're collecting good enough info here